### PR TITLE
PHP 8.1 support

### PR DESCRIPTION
**Pull request by arueckauer**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Drops support for PHP versions prior to 7.4
- Adds support for PHP version 8.1
- Upgrades
  - `laminas/laminas-coding-standard` from `1.0.0` to latest `2.3.0`
  - `phpunit/phpunit` to latest `9.5.13`
  - `vimeo/psalm` to latest `4.20.0`

### CODE REVIEW

**Comment by Ocramius** [comment]
You know it's coming: phpcs upgrade 😬

**Comment by Ocramius** [comment]
Looks like the `3.5.x` branch doesn't exist 😩

**Comment by arueckauer** [comment]
> You know it's coming: phpcs upgrade 😬

I do - working on it. If that just would be the only thing! 🙈 Strict types might give this some hickups.

File: src/Smd/Service.php [hunk 1]
```diff
@@ -166,10 +182,8 @@ public function setName($name)
 
     /**
      * Retrieve name.
-     *
-     * @return string
      */
-    public function getName()
+    public function getName(): string
```
Comment (Ocramius) [inline src/Smd/Service.php]: Careful: adding return types to open classes is a BC break - please avoid doing this on non-`final` classes

File: src/Smd/Service.php [hunk 2]
```diff
@@ -221,10 +235,8 @@ public function setTarget($target)
 
     /**
      * Get service target.
-     *
-     * @return string
      */
-    public function getTarget()
+    public function getTarget(): ?string
```
Comment (Ocramius) [inline src/Smd/Service.php]: Careful: adding return types to open classes is a BC break - please avoid doing this on non-`final` classes

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
I deleted the `4.0.0` milestone, made a backup of the `4.0.x` branch, and deleted `4.0.x` too.

This patch is now re-targeted onto `3.5.x`

File: src/Smd/Service.php (hunk 2 above)
Comment (arueckauer) [reply src/Smd/Service.php]: Agreed, reverted.

File: src/Smd/Service.php (hunk 1 above)
Comment (arueckauer) [reply src/Smd/Service.php]: Agreed, reverted.

**Review by Ocramius — APPROVED** [review]
🚢  thanks @arueckauer!

**Pull request merged**